OpenRA.Mods.Common/Pathfinder/PathGraph.cs

```diff
@@ -94,17 +94,27 @@ sealed class PathGraph : IGraph<CellInfo>
 		readonly Dictionary<byte, (ICustomMovementLayer Layer, CellLayer<CellInfo> Info)> customLayerInfo =
 			new Dictionary<byte, (ICustomMovementLayer, CellLayer<CellInfo>)>();
 
+		ICustomMovementLayer[] customMovementLayers;
+
 		public PathGraph(CellInfoLayerPool layerPool, Locomotor locomotor, Actor actor, World world, BlockedByActor check)
 		{
 			pooledLayer = layerPool.Get();
 			groundInfo = pooledLayer.GetLayer();
 			var locomotorInfo = locomotor.Info;
 			this.locomotor = locomotor;
-			var layers = world.GetCustomMovementLayers().Values
-				.Where(cml => cml.EnabledForActor(actor.Info, locomotorInfo));
 
-			foreach (var cml in layers)
-				customLayerInfo[cml.Index] = (cml, pooledLayer.GetLayer());
+			// PERF: Avoid LINQ
```

Comment:
These changes feel a bit too ugly IMO.

None of the `EnabledForActor` implementations actually use the `ActorInfo`, so an alternative that could save even more calculations could be to change that to `EnabledForLocomotor` and then calculate and store the enabled layers once in each locomotor. We can then use them directly in each PathGraph.

Reply:
Can we make that a separate future pull request.

In the mean time we could keep these lines or use the original code. Let me know if you want me to revert these lines back.

Reply:
Ok lets revert these back and leave this to a followup PR.

Reply:
Can we get these changes removed from this PR so we can move forward?

Reply:
The array that caches the hashmap were reverted in the last push. What is left was the original code.